text/XXXX-delete_and_drop.md

```diff
@@ -0,0 +1,225 @@
+1. rename `remove` to `delete`, clarify its text to talk about history instead of just storage
+2. Add an `amend` flag on the `save` command that _replaces_ the HEAD commit
+3. Add a new top-level command: `drop`, for dropping datasets you don't own
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+### Renaming Remove
+Here's the current (qri v0.9.8) description of the `remove` command:
+
+```
+$ qri remove --help
+Remove gets rid of a dataset from your qri node. After running remove, qri will
+no longer list your dataset as being available locally. 
+[rambling bit about IPFS storage redacted]
+```
+
+This text is misleading. It seems to be about storage, when the primary purpose of  `remove` is to edit history. Storage savings are a byproduct of editing one or more commits out of a history. This RFC sets aside the tight coupling between history and storage for another day, and instead focueses on fixing the role of remove by clearifying it's documentation:
+
+```
+$ qri delete --help
+Delete removes commits from a dataset history. Use delete to "correct the record", 
+removing commits in a dataset history. Delete always works from the latest (HEAD) 
+commit, working backwards toward the first commit. To delete from a dataset you 
+must be able to edit it.
+
+Delete requires a "revisions" flag, specifying
+
+To delete an entire dataset and free up the name again, use `--all` (or a 
+number of revisions longer than the history itself). Before doing this be sure
+to drop your dataset from other remotes
+
+Usage:
+  qri delete [DATASET] [--revisions|--all] [flags]
+
+Examples:
+  # delete the latest commit from annual_pop
+  $ qri delete me/annual_pop --revisions 1
+
+  # delete the latest two versions from history, keeping the underlying data
+  $ qri delete me/annual_pop --revisions 2  --keep-storage
+
+  # destroy a dataset named `annual_pop`
+  $ qri delete --all me/annual_pop
+
+Flags:
+   --keep-storage     don't remove the deleted versions from storage
+
+```
+
+"Delete" sounds more destructive than "remove", and that's a good thing in this case, becuase delete is editing history. The delete command is also how you _completely remove_ a dataset (with the `--all` flag). The simple change of having a delete command exist should help users find a way to edit history.
+
+### Amend
+
+Running `save` with the `--amend` flag replaces the current HEAD with a new commit. This is pulled directly from git. 
+
+Adding an `--amend` flag has the UX advantage of working language that resembles "undo" directly into the save command, which users will be familiar with before they ever need `delete`. As a git user I (b5) learned to use `--amend` before `rebase`, which I admit wasn't great, but allowed me to stick with git long enough to understand what rebase did.
+
+Users should come to think of `--amend` as a shortcut:
+
+```
+# two commands:
+$ qri delete me/dataset --revisions 1
```

Comment:
Can `--amend` only squash 1 revision? I'm fine if that's the case, just want to make sure. It certainly makes it easier to implement.

If the user has already pushed their head, do we detect that and fail with an error? Or let them implicitly branch and deal with it later somehow?

Reply:
yes amend can only ever squash one commit.


The `amend` will be marked in logbook. When pushing after amending the receiving remote will update and remove the orphaned commit, or at least know they _can_ remove the orphaned commit. 

They don't have the option to branch, but we could later build tools that show more detailed histories, allowing a user to see dropped commits still recorded in logbook.